src/main/java/com/solace/samples/java/patterns/GuaranteedBlockingPublisher.java

```diff
@@ -0,0 +1,152 @@
+        }
+        System.out.println(API + " " + SAMPLE_NAME + " initializing...");
+
+        final Properties properties = new Properties();
+        properties.setProperty(TransportLayerProperties.HOST, args[0]);          // host:port
+        properties.setProperty(ServiceProperties.VPN_NAME,  args[1]);     // message-vpn
+        properties.setProperty(AuthenticationProperties.SCHEME_BASIC_USER_NAME, args[2]);      // client-username
+        if (args.length > 3) {
+            properties.setProperty(AuthenticationProperties.SCHEME_BASIC_PASSWORD, args[3]);  // client-password
+        }
+        //properties.setProperty(JCSMPProperties.GENERATE_SEQUENCE_NUMBERS, true);  // not required, but interesting
+        properties.setProperty(TransportLayerProperties.RECONNECTION_ATTEMPTS, "20");  // recommended settings
+        properties.setProperty(TransportLayerProperties.CONNECTION_RETRIES_PER_HOST, "5");
+        // https://docs.solace.com/Solace-PubSub-Messaging-APIs/API-Developer-Guide/Configuring-Connection-T.htm
+        // ready to connect now
+        final MessagingService messagingService = MessagingService.builder(ConfigurationProfile.V1)
+                .fromProperties(properties)
+                .build();
+        messagingService.connect();  // blocking connect
+        messagingService.addServiceInterruptionListener(serviceEvent -> {
+            logger.info("### SERVICE INTERRUPTION: "+serviceEvent.getCause());
+            //isShutdown = true;
+        });
+        messagingService.addReconnectionAttemptListener(serviceEvent -> {
+            logger.info("### RECONNECTING ATTEMPT: "+serviceEvent);
+        });
+        messagingService.addReconnectionListener(serviceEvent -> {
+            logger.info("### RECONNECTED: "+serviceEvent);
+        });
+        
+        // build the publisher object, starts its own thread
+        final PersistentMessagePublisher publisher = messagingService.createPersistentMessagePublisherBuilder()
+                .withDeliveryAckWindowSize(30)
+                .build();
+        publisher.start();
+        
+        ScheduledExecutorService statsPrintingThread = Executors.newSingleThreadScheduledExecutor();
+        statsPrintingThread.scheduleAtFixedRate(() -> {
+            System.out.printf("%s %s Published msgs/s: %,d%n",API,SAMPLE_NAME,msgSentCounter);  // simple way of calculating message rates
+            msgSentCounter = 0;
+        }, 1, 1, TimeUnit.SECONDS);
+        
+        System.out.println(API + " " + SAMPLE_NAME + " connected, and running. Press [ENTER] to quit.");
+        System.out.println("Publishing to topic '"+ TOPIC_PREFIX + API.toLowerCase() + 
+                "/pers/pub/...', please ensure queue has matching subscription."); 
+        byte[] payload = new byte[PAYLOAD_SIZE];  // preallocate memory, for reuse, for performance
+        Properties messageProps = new Properties();
+        messageProps.put(MessageProperties.PERSISTENT_ACK_IMMEDIATELY, "true");
```

Comment:
This is a bug in the current 1.0 version... when using a blocking publish, the API should automatically set this per-message property, so the broker doesn't wait 1 second to reply with the ACK since the AD publish window is automatically set to 255 by default in the new API.  The broker waits until 1/3rd of the window is full before sending an ACK, or after 1 second has elapsed.  I'll remove it for now, so we don't have to update it when 1.1 comes along, but it means this sample will only publish 1 message per second.  Thoughts?  Or I leave it in and take it out when 1.1 is released..?

Reply:
leave it in with a comment to remove when that bug is fixed. Also please post a link to it on the bug in JIRA so it can be fixed. Just say "Please make a PR removing this 1 line before closing"